ansible/roles/agnosticd_virtualenv/tasks/create.yml

```diff
@@ -0,0 +1,25 @@
+---
+- name: Create temporary directory for {{ agnosticd_virtualenv }}
+  tempfile:
+    path: "{{ agnosticd_virtualenv_base_dir }}"
+    prefix: agnosticd-venv-
+    suffix: -tmp
+    state: directory
+  register: r_agnosticd_virtualenv_tmp
+
+- name: Install packages into agnosticd virtualenv tmp
+  pip:
+    name: "{{ agnosticd_virtualenv_requirements }}"
+    virtualenv: "{{ r_agnosticd_virtualenv_tmp.path }}"
+    virtualenv_python: "{{ agnosticd_virtualenv_python }}"
+
+# Ansible has no file move or rename module, so command will have to do
+- name: Move agnosticd virtualenv tmp to {{ agnosticd_virtualenv }}
+  command: mv {{ r_agnosticd_virtualenv_tmp.path }} {{ agnosticd_virtualenv }}
+  # Failed to mv virtualenv most likely means that another agnosticd run created it first
+  failed_when: false
```

Comment:
Why do you create a temp directory and don't work directly in the target dir `agnosticd_virtualenv` ?

Reply:
There is a race condition where two ansible runs on the same host that will use the same virtualenv start around the same time. Whichever one gets there second may see the virtualenv directory exists and proceed when it is actually still creating, so this is a safe design pattern to make sure the creation of the virtualenv is atomic and avoid errors if we do end up in this situation.

Reply:
OK, that's what i guess,  but i thought it would be very unlikely.

> may see the virtualenv directory exists and proceed 

That would not be too much of a problem, only for that consecutive simultaneous run, the future runs would be fine.
On the other hand, I guess 2 simultaneous runs could build the virtualenv at the same time and could leave a broken virtualenv. That would be annoying.

I was not really sure if it was worth the additional tasks, but i guess now that it's written.. :)